std/haxe/macro/ExprTools.hx

```diff
@@ -128,6 +128,11 @@ class ExprTools {
 				}
 				if (edef != null && edef.expr != null)
 					f(edef);
+			case EFormat(parts):
+				for (part in parts) switch part.kind {
+					case FRaw(_) | FIdent(_):
```

Comment:
It's usually a good idea to design your data in a way that it doesn't lose information. Without FIdent, the cases `$ident` and `${ident}` would be indistinguishable and could not be restored from the data. This would certainly be a concern for hxparser. Maybe not so much for the compiler parser because it mostly "doesn't matter", but I still think this is the better approach.